stress/v2/ponyExpress/util.go

```diff
@@ -0,0 +1,88 @@
+package ponyExpress
+
+import (
+	"crypto/rand"
+	"fmt"
+	"strconv"
+	"sync"
+)
+
+// ###########################################
+// ConcurrencyLimiter and associated methods #
+// ###########################################
+
+// ConcurrencyLimiter ensures that no more than a specified
+// max number of goroutines are running.
+type ConcurrencyLimiter struct {
+	inc   chan chan struct{}
+	dec   chan struct{}
+	max   int
+	count int
+
+	sync.Mutex
+}
+
+// NewConcurrencyLimiter returns a configured limiter that will
+// ensure that calls to Increment will block if the max is hit.
+func NewConcurrencyLimiter(max int) *ConcurrencyLimiter {
+	c := &ConcurrencyLimiter{
+		inc: make(chan chan struct{}),
+		dec: make(chan struct{}, max),
+		max: max,
+	}
+	go c.handleLimits()
+	return c
+}
+
+// Increment will increase the count of running goroutines by 1.
+// if the number is currently at the max, the call to Increment
+// will block until another goroutine decrements.
+func (c *ConcurrencyLimiter) Increment() {
+	r := make(chan struct{})
+	c.inc <- r
+	<-r
+}
+
+// Decrement will reduce the count of running goroutines by 1
+func (c *ConcurrencyLimiter) Decrement() {
+	c.dec <- struct{}{}
+}
+
+// NewMax resets the max of a ConcurrencyLimiter.
+func (c *ConcurrencyLimiter) NewMax(i int) {
+	c.Lock()
+	defer c.Unlock()
+	c.max = i
+}
+
+// handleLimits runs in a goroutine to manage the count of
+// running goroutines.
+func (c *ConcurrencyLimiter) handleLimits() {
+	for {
+		r := <-c.inc
+		c.Lock()
+		if c.count >= c.max {
+			<-c.dec
+			c.count--
+		}
+		c.Unlock()
+		c.count++
+		r <- struct{}{}
+	}
+}
+
+// Utility interger parsing function
+func parseInt(s string) int {
+	i, err := strconv.ParseInt(s, 10, 64)
+	if err != nil {
+		fmt.Errorf("Error parsing integer:\n  String: %v\n  Error: %v\n", s, err)
```

Comment:
What are you intending to do with this error? Maybe you wanted to log it or panic?